test/integration/vectorSearch.test.ts

```diff
@@ -39,59 +39,69 @@ afterEach(async (ctx) => {
 
 describe('search', () => {
   test('search 1 2 3 4', async () => {
-    const results = await xata.db.users.vectorSearch('vector', [1, 2, 3, 4]);
+    const { records: results, totalCount } = await xata.db.users.vectorSearch('vector', [1, 2, 3, 4]);
 
+    expect(totalCount).toEqual(4);
     expect(results.map((r) => r.full_name)).toEqual(['r4', 'r1', 'r2', 'r3']);
   });
 
   test('search 0.4 0.3 0.2 0.1', async () => {
-    const results = await xata.db.users.vectorSearch('vector', [0.4, 0.3, 0.2, 0.1]);
+    const { records: results, totalCount } = await xata.db.users.vectorSearch('vector', [0.4, 0.3, 0.2, 0.1]);
 
+    expect(totalCount).toEqual(4);
     expect(results.map((r) => r.full_name)).toEqual(['r2', 'r3', 'r4', 'r1']);
   });
 
   test('with size', async () => {
-    const results = await xata.db.users.vectorSearch('vector', [1, 2, 3, 4], { size: 2 });
+    const { records: results, totalCount } = await xata.db.users.vectorSearch('vector', [1, 2, 3, 4], { size: 2 });
 
+    expect(totalCount).toEqual(4);
```

Comment:
Worth noting that the backend seems to return the total count of results regardless of the page size specified, so I matched against 4 here rather than 2

Reply:
Please add a comment